src/SwedbankPay.Sdk/Payments/SwishPayments/PaymentRequestObject.cs

```diff
@@ -0,0 +1,56 @@
+﻿using Newtonsoft.Json;
+
+using System.Collections.Generic;
+using System.Globalization;
+
+namespace SwedbankPay.Sdk.Payments.SwishPayments
+{
+    public class PaymentRequestObject
+    {
+        protected internal PaymentRequestObject(CurrencyCode currency,
+                                                List<Price> prices,
+                                                string description,
+                                                string payerReference,
+                                                string userAgent,
+                                                CultureInfo language,
+                                                Urls urls,
+                                                PayeeInfo payeeInfo,
+                                                PrefillInfo prefillInfo,
+                                                SwishRequest swishRequest,
+                                                Dictionary<string, object> metaData = null)
```

Comment:
Arguments with `null` default value can just be removed from the constructor and have their property changed to a `public` setter instead. Only required, non-default arguments should exist in the constructor.

Reply:
Isn't this against the Immutable principle?

Reply:
I agree with zunkas. Plus having to mix both makes list initialization annoying. Either all as properties or all as constructor, don't mix both if we don't have to.

Reply:
The way it is now we end up with enormous constructors, which is less than ideal. I know @muzukashiine had issues with this as well. The greatest value of constructors is being able to validate the arguments to ensure that they follow our business rules of not being `null`, having invalid data, etc., before an instance of a class can be constructed. Once constructed, that instance should follow the business rules. If a class has a lot of optional fields, having them as constructor arguments provide no real value, as there is no validation that can be performed on them.

Reply:
My opinion is that a bunch of arguments should not be passed to the constructor, only to be continually passed down. Group arguments into objects as early as possible and pass in these instead. Example: A complex `Person` class would not have 100 arguments if there were 100 details one would want to describe about a person. These would come grouped into objects that would themselves be grouped again and again and so on. My rule of thumb is if an argument list grows past X elements I need to think about how I can group them better.

Having both constructor arguments and public setters uses two different styles. I would prefer we stick to one, like mentioned before by NullableInt. However, I also agree that passing in optional arguments to the constructor can feel bloaty and unnecessary. But I'm not a fan of using public setters either, I much rather we use stricter setters (aka methods with clear business intent). That's what I would have done.